aeron-archive/src/main/java/io/aeron/archive/Catalog.java

```diff
@@ -656,6 +656,108 @@ void extendRecording(
         forceWrites(catalogChannel);
     }
 
+    void replaceRecording(
+        final long recordingId,
+        final long startPosition,
+        final long stopPosition,
+        final long startTimestamp,
+        final long stopTimestamp,
+        final int imageInitialTermId,
+        final int segmentFileLength,
+        final int termBufferLength,
+        final int mtuLength,
+        final int sessionId,
+        final int streamId,
+        final String strippedChannel,
+        final String originalChannel,
+        final String sourceIdentity)
+    {
+        final int recordingOffset = recordingDescriptorOffset(recordingId);
+        if (-1 == recordingOffset)
+        {
+            throw new ArchiveException("unknown recording id: " + recordingId);
+        }
+
+        final int recordingLength = fieldAccessBuffer.getInt(
+            recordingOffset + RecordingDescriptorHeaderDecoder.lengthEncodingOffset(), BYTE_ORDER);
+        final int oldFrameLength = align(recordingLength + DESCRIPTOR_HEADER_LENGTH, alignment);
+        final int newFrameLength = recordingDescriptorFrameLength(strippedChannel, originalChannel, sourceIdentity);
+        final int length, checksumLength;
+        if (newFrameLength > oldFrameLength)
+        {
+            length = checksumLength = newFrameLength - DESCRIPTOR_HEADER_LENGTH;
+
+            final int numExtraBytes = newFrameLength - oldFrameLength;
+            final int endOfLastRecording = nextRecordingDescriptorOffset;
+            if (endOfLastRecording + numExtraBytes > capacity)
+            {
+                growCatalog(MAX_CATALOG_LENGTH, numExtraBytes);
+            }
+
+            nextRecordingDescriptorOffset += numExtraBytes;
+
+            final long[] index = catalogIndex.index();
+            final int lastPosition = catalogIndex.lastPosition();
+            if (recordingId != index[lastPosition])
+            {
+                //  if not the last recording then shift existing data to the end of the file
+                fieldAccessBuffer.putBytes(
+                    recordingOffset + newFrameLength,
+                    fieldAccessBuffer,
+                    recordingOffset + oldFrameLength,
+                    endOfLastRecording - (recordingOffset + oldFrameLength));
+
+                // fixup the index of the shifted data
```

Comment:
Avoid use of comments by extracting blocks to named methods.

Reply:
Done.